### DISCUSSION TITLE
Handle redirect to Unix Domain socket

### DISCUSSION THREAD

**Discussion by orodbhen** [projectdiscovery/httpx#1262] [Q&A] [answered]
I have a multi-service container application, each of which listens for requests on a unix socket. I'll like to have the client pass all requests to the central service, and have it redirect to the others. but I haven't been able to find a way to do that. When the redirect comes back from the server it raises an `UnsupportedProtocol` error for `unix://`.

I used the `uds` parameter for `HTTPTransport` to establish the initial connections, but I'm not sure if there's a way to add transports dynamically. I've looked at implementing a custom transport, and examined the source code for `HTTPTransport`, and it seems like this would be very complex.

Is there a straightforward way of doing this that I'm not seeing? It would be great to have something like a `UDSTransport` class to handle this scenario.

One option would be to use a mount map, and have only the central service use a unix socket, but I'd prefer to have them all do that.

**Comment by olearycrew** [comment]
I think you're looking for the python httpx library?

https://www.python-httpx.org/

**Reply by orodbhen** [reply, marked as answer]
So I am. Thanks.
